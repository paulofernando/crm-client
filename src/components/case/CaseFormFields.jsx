import React from "react";
import { Form, InputGroup } from "react-bootstrap";
import DatePicker from "react-datepicker";
import "react-datepicker/dist/react-datepicker.css";

import "../../App.css";

class CaseFormFields extends React.Component {
  render() {
    const {
      values,
      errors,
      touched,
      handleChange,
      handleBlur,
      handleDateChange,
      courtDate
    } = this.props;

    return (
      <>
        <Form.Group controlId="formTitle">
          <Form.Label>Title *</Form.Label>
          <Form.Control
            type="text"
            name="title"
            className="formTitle"
            onChange={handleChange}
            onBlur={handleBlur}
            value={values.title}
            className={touched.title && errors.title ? "error" : null}
          />
          {touched.title && errors.title ? (
            <div className="error-message">{errors.title}</div>
          ) : null}
        </Form.Group>
        <Form.Group controlId="formLastName">
          <Form.Label>Description</Form.Label>
          <Form.Control
            as="textarea"
            name="description"
            onChange={handleChange}
            onBlur={handleBlur}
            value={values.description}
            className={
              touched.description && errors.description ? "error" : null
            }
          />
          {touched.description && errors.description ? (
            <div className="error-message">{errors.description}</div>
          ) : null}
        </Form.Group>
        <Form.Group style={{ width: "50%" }} controlId="formEmail">
          <Form.Label>Value *</Form.Label>
          <InputGroup>
            <InputGroup.Prepend>
              <InputGroup.Text>$</InputGroup.Text>
            </InputGroup.Prepend>
            <Form.Control
              type="text"
              name="value"
              onChange={handleChange}
              onBlur={handleBlur}
              value={values.value}
              className={touched.value && errors.value ? "error" : null}
              style={{ textAlign: "right" }}
            />
            <InputGroup.Append>
              <InputGroup.Text>.00</InputGroup.Text>
            </InputGroup.Append>
          </InputGroup>
          {touched.value && errors.value ? (
            <div className="error-message">{errors.value}</div>
          ) : null}
        </Form.Group>
        <Form.Group style={{ width: "50%" }} controlId="formCourtDate">
          <Form.Label>Court Date *</Form.Label>
          <DatePicker
            name="courtDate"
            onBlur={handleBlur}
            selected={courtDate}
            onChange={handleDateChange}
            className={touched.courtDate && errors.courtDate ? "error" : null}
          />
          {touched.courtDate && !courtDate ? (
            <div className="error-message">This field is required</div>
          ) : null}
        </Form.Group>
      </>
    );
  }
}

export default CaseFormFields;
