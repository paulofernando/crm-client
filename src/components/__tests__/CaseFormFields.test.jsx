import React from 'react';
import renderer from 'react-test-renderer';

import CaseFormFields from '../case/CaseFormFields';

// -------------------------------

const courtCase = {
    title: "Title test",
    description: "Description test",
    value: 1000
}

const errors = {
    title: ""
}

const touched = {
    title: ""
}

const CaseFormFieldsComponent = (
    <CaseFormFields
        values={courtCase}
        errors={errors}
        touched={touched}
        courtDate={new Date("2020-01-01T00:00:00.000Z")} />
)

// -------------------------------

describe('CaseFormFields', () => {
    it('renders correctly', () => {
        const wrapper = renderer
            .create(CaseFormFieldsComponent)
            .toJSON();
        expect(wrapper).toMatchSnapshot();
    });

});