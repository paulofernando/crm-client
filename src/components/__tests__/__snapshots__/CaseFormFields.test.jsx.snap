// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CaseFormFields renders correctly 1`] = `
Array [
  <div
    className="form-group"
  >
    <label
      className="form-label"
      htmlFor="formTitle"
    >
      Title *
    </label>
    <input
      className="form-control"
      id="formTitle"
      name="title"
      type="text"
      value="Title test"
    />
  </div>,
  <div
    className="form-group"
  >
    <label
      className="form-label"
      htmlFor="formLastName"
    >
      Description
    </label>
    <textarea
      className="form-control"
      id="formLastName"
      name="description"
      value="Description test"
    />
  </div>,
  <div
    className="form-group"
    style={
      Object {
        "width": "50%",
      }
    }
  >
    <label
      className="form-label"
      htmlFor="formEmail"
    >
      Value *
    </label>
    <div
      className="input-group"
    >
      <div
        className="input-group-prepend"
      >
        <span
          className="input-group-text"
        >
          $
        </span>
      </div>
      <input
        className="form-control"
        id="formEmail"
        name="value"
        style={
          Object {
            "textAlign": "right",
          }
        }
        type="text"
        value={1000}
      />
      <div
        className="input-group-append"
      >
        <span
          className="input-group-text"
        >
          .00
        </span>
      </div>
    </div>
  </div>,
  <div
    className="form-group"
    style={
      Object {
        "width": "50%",
      }
    }
  >
    <label
      className="form-label"
      htmlFor="formCourtDate"
    >
      Court Date *
    </label>
    <div
      className="react-datepicker-wrapper"
    >
      <div
        className="react-datepicker__input-container"
      >
        <input
          className=""
          disabled={false}
          name="courtDate"
          onBlur={[Function]}
          onChange={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          readOnly={false}
          type="text"
          value="12/31/2019"
        />
      </div>
    </div>
  </div>,
]
`;
